src/Equinor.ProCoSys.DbSync/ValueConversion.cs

```
@@ -0,0 +1,182 @@
+﻿using static Equinor.ProCoSys.Completion.DbSyncToPCS4.SyncMappingConfig;
+
+namespace Equinor.ProCoSys.Completion.DbSyncToPCS4
+{
+    public static class ValueConversion
+    {
+        /**
+         * Returns a string representing the target sql parameter value to be included in the sql statement. 
+         * If the column config includes a conversion method, this will be used. If not, standard conversion will be used. 
+         */
+        public static async Task<string> GetSqlParameterValueAsync(object? sourceValue, ColumnSyncConfig columnConfig, IOracleDBExecutor oracleDBExecutor, CancellationToken cancellationToken)
+        {
+            if (columnConfig.ValueConversionMethod != null)
+            {
+                return await ConvertToSqlParameterValueAsync(sourceValue, columnConfig.ValueConversionMethod, oracleDBExecutor, cancellationToken);
+            }
+
+            switch (columnConfig.SourceType)
+            {
+                case DataColumnType.String:
+                    return sourceValue != null ? $"'{sourceValue}'" : "null";
+                case DataColumnType.Int:
+                    return sourceValue != null ? $"{sourceValue}" : "null";
+                case DataColumnType.Bool:
+                    if (sourceValue == null || sourceValue.ToString() == "0")
```

This assumes null == false. This is not always the case.

---

Good catch! Was uncertain about this. I assume that boolean columns will allways be 'non null'? Maybe we should just throw an exception here?